pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -332,6 +346,12 @@ protected void handleReachedEndOfTopic(CommandReachedEndOfTopic commandReachedEn
     private boolean addPendingLookupRequests(long requestId, CompletableFuture<LookupDataResult> future) {
         if (pendingLookupRequestSemaphore.tryAcquire()) {
             pendingLookupRequests.put(requestId, future);
+            eventLoopGroup.schedule(() -> {
+                if (!future.isDone()) {
+                    future.completeExceptionally(new TimeoutException(
+                            requestId + " lookup request timedout after ms " + DEFAULT_CONNECT_TIMEOUT_IN_SECONDS));
+                }
+            }, DEFAULT_CONNECT_TIMEOUT_IN_SECONDS, TimeUnit.SECONDS);
```

Comment:
I think we should be really using the `operationTimeout` that is configurable on the Client itself.